internal/pointer/summaries.go

```diff
@@ -0,0 +1,32 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
```

Comment:
I'm not sure how different licenses work within the same codebase so we may need to remove our copyright header here

Reply:
I think since this a a new file this is fine. We should include the BSD license, maybe in the pointer folder though (for the Go authors) and include "modified by ..." when we modify the pointer files. I'm not sure how to do the latter yet.